homeassistant/components/homekit_controller/connection.py

```diff
@@ -182,14 +180,10 @@ def async_set_available_state(self, available: bool) -> None:
 
     async def async_setup(self) -> None:
         """Prepare to use a paired HomeKit device in Home Assistant."""
-        entity_storage: EntityMapStorage = self.hass.data[ENTITY_MAP]
         pairing = self.pairing
         transport = pairing.transport
         entry = self.config_entry
 
-        if cache := entity_storage.get_map(self.unique_id):
```

Comment:
aiohomekit will have restored its cache directly with the aiohomekit changes.

Reply:
Will the cache exist or do we need to restore it once from HA?

Reply:
The cache will exist. `entity_storage` implements `aiohomekit.characteristic_cache.CharacteristicCacheType` and will already be passed to the controller as part of `async_get_controller` (this bit was missing, thought it already existing but just realised it was missing). So the call to `_load_accessories_from_cache` in aiohomekit will load from the same cache as this one.